src/xdocs-test/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java

```diff
@@ -0,0 +1,19 @@
+/*
+AbstractClassName
+format = (default)^Abstract.+$
+ignoreModifier = (default)false
+ignoreName = (default)false
```

Comment:
Is it a future plan to also assert the config in the xdoc must match this in the example?

Reply:
Only this check and other 2 more complicated ones(`ConstantName` and `VisibilityModifier`) are getting "new" xdoc examples. The rest would just be translated to tests without changing the xdoc. In the future there will be a parser that will extract the input config(and turn it into xml) and java code and place them in the xdoc. There will be placeholders for those.

There is an issue to discuss what the parser will do exactly https://github.com/checkstyle/checkstyle/issues/13134.

Reply:
@rnveach , there will no assert on config. As we will use config from example as is.
It will be easy to catch during review that proposed config is weird, quality of config is human interpretation.

Example java file has both config  + java. Regular test phase execute it to make sure it is compileable and violations lines match. During xdoc generation, we parse Example and split into two parts: config, java snippet( no package, no imports)

Reply:
> In the future there will be a parser that will extract the input config(and turn it into xml) and java code and place them in the xdoc
> During xdoc generation, we parse Example and split into two parts: config, java snippet

So there is basically a plan for an assertion that xdoc config must match example config, as example config will be inserted into xdoc.

Right now there is no assertion and so it is on reviewers to manually review the 2 match up.

That is all I was asking.

Reply:
> So there is basically a plan for an assertion that xdoc config must match example config, as example config will be inserted into xdoc

No. Xdoc example content will be generated from the example tests. Xdocs examples will contain placeholders only that tell automation where the examples should be located.

Reply:
We will not have matching anymore.
We will use it as-is from Example.java files.
Example files are 100% Input files, but designed for users to see on website ,not a test coverage.

Mismatch of actual violations vs declared inside as trailing comments - will be test phase failure ( same as we do for Inputs)

Reply:
After thinking more on https://github.com/checkstyle/checkstyle/pull/13112#discussion_r1217239328, I think we are missing the point. Even after we begin to automate xdoc generation process (while we transition existing docs), we still count on reviewers to ensure that content matches up. @rnveach is this what you are getting at?

Reply:
I believe so.

The issue I see right now is there is no validation that things synch up between the old and new areas.

We have the XML configuration in the XDOC right now ( https://github.com/checkstyle/checkstyle/blob/a1dbb5c450bc9fcf6249676a773ac3fcdbed9a6d/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractClassNameCheck.java#L63 ), and we have a Javadoc of a "like" configuration in the Java example file ( https://github.com/checkstyle/checkstyle/blob/a1dbb5c450bc9fcf6249676a773ac3fcdbed9a6d/src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/naming/abstractclassname/Example1.java#L2-L5 ) .

How are we going to verify the 2 always match up and that the PR author didn't accidentally swap around and put in a Javadoc configuration for the wrong XML configuration?

I see a hole here where you have things that need to match up to make sense and there is no matching right now, and you want to rewrite probably hundreds of examples by hand which can be error prone. I see something going wrong since you will require reviewers to look at every nook and cranny.

If you want to see what this will be like for a few examples, then ok. But lets say "yah or nah" and add some automation sooner so we don't make a mess our documentation. We use to enable automation for a few parts and then expand it as we go.

Reply:
All examples already not good, as they are crafted manually, and we already see facts that comments are not matching with actual behavior.

Such 3 Check updates are just to explicitly place all of us on same page, how it will looks like for user, validaitons are out of scope for now and it is much better to do them at the end as we see all nuances of parser, generator, .... .

we already have idea to allow xml config - https://github.com/checkstyle/checkstyle/issues/12155 in Examples.
But before it, we use what we have - properties.

YES, we will not have supper ideal solution to control all and somebody can put in template file config from example1 and java from example2. But this is easy to catch during review and fix even if it reported by user after release.

Goal is:
Javadoc of Check completely loose examples. Javadoc of Check own only metadata.
Examples are separate entities like Input tests but slightly different.
Each Check have it own template file for xdoc, to let our generator fill it with data (search/replace).

Reply:
> If you want to see what this will be like for a few examples, then ok.

only 3 PRs like this. All others will be based on some automation.
if we ever endup with controversial module, we can do one more module manually to make sure we agree on final result again, and do not mess with ire-mplementation of automation.

Reply:
I added a task for writing a script to ensure examples match with xdocs at https://github.com/orgs/checkstyle/projects/5.

Reply:
Be careful with matching, I can imagine that majority of examples are not actual or not compileable or ....

Reply:
@stoyanK7 As the discussion in discord, should these PRs add the ID to connect the xdoc and input together?

Reply:
@rnveach I opened https://github.com/checkstyle/checkstyle/issues/13205 for this. To not forget to add ids here, I changed from `AbstractClassName` to `AbstractClassName (needs anchors)` in this issue's checklist.

> Also, I will not be validating the conversion process in these PRs (java and config were transferred correctly and it is still the same example). I am expecting @romani and/or @nrmancuso to do that as it is too manual.

No problem :)